scanpy/plotting/_tools/scatterplots.py

```diff
@@ -741,6 +734,7 @@ def spatial(
     alpha_img: float = 1.0,
     bw: bool = False,
     size: float = None,
+    na_color=(0, 0, 0, 0),
```

Comment:
@flying-sheep, the signature for `sc.pl.spatial` is being generated incorrectly. The documented default value is the value from `embedding`, not what's defined here.

```python
?sc.pl.spatial
...
    na_color: Any = 'lightgray',
```

It looks like this has to do with how `inspect` defines signatures. By default, if a function is wrapped, it skips the wrapper and just gets the signature of the wrapped object. If you turn that off, it only looks at the signature of the wrapper. What we want is the original signature updated with the wrapping signature.

I think this can be fixed in getdoc pretty easily, but this does not fix the signature you see in ipython. Ideally this could be fixed at the wrapping level, but I haven't figured out how yet. It looks like `inspect` kinda handles this for `partial`s in `_signature_get_partial`, but that can't be used as is here since it relies on the attributes of partial functions.

Any suggestions on how to fix this short of defining a modified `_signature_get_partial` and setting the `__signature__` attribute on these?

Reply:
No worries @giovp, this is something @flying-sheep set up so I think he's probably in the best place to understand it.

Reply:
Made it work by defining a `__signature__` and removing `__wrapped__`. 

`__signature__ ` make `getdoc` work correctly, while removing `__wrapped__` made the sphinx generated documentation have the correct default values.